flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/ContaineredTaskManagerParameters.java

```diff
@@ -158,8 +158,10 @@ public static ContaineredTaskManagerParameters create(
 
 		// (2) split the remaining Java memory between heap and off-heap
 		final long heapSizeMB = TaskManagerServices.calculateHeapSizeMB(containerMemoryMB - cutoffMB, config);
-		// use the cut-off memory for off-heap (that was its intention)
-		final long offHeapSizeMB = containerMemoryMB - heapSizeMB;
+		// (3) try to compute the offHeapMemory from a safe margin
+		final long restMemoryMB = containerMemoryMB - heapSizeMB;
+		final long offHeapCutoffMemory = calculateOffHeapCutoffMB(config, restMemoryMB);
```

Comment:
Thank you for your attention. I do add a containerized.offheap-cutoff-ratio for two reason
 1.  it will be unfriendly for  a small memory container such as 1gb memory container  if we only use containerized.heap-cutoff-ratio and containerized.heap-cutoff-min to cutoff the memory 0f container physical memory .  
 2. native memory  not only  contains memory which used by rockdb state backend but also   memory pools are called arenas which will cost enormous memory in multi-core cpu machine
as time goes on . You can use 'pmap -x {TM container pid}' command to validate it 
 So I want to  add a  option to  adjust memory  flexiblely  for someone who wana adjust both heap meomory and offheap memory to adapt the three kinds of memory.

Reply:
Thanks for your suggestion.  It maybe  bring  more refactor  to the project  if  we name the current "containerized.heap-cutoff-ratio" to "containerized.memory-cutoff-ratio".Because we need to adapt  a small memory container due to 600 mega byte of the current "containerized.heap-cutoff-min". How about we set a  common default value of "containerized.offheap-cutoff-ratio" and "containerized.offheap-cutoff-min" so that user do not need to care the two parameters at most condition.

Reply:
Sorry I have not got the point of more refactor you mentioned. I think you just need to adjust both parameters of `containerized.memory-cutoff-ratio` and `containerized.memory-cutoff-min` for your project. 

Unless in your case, modify the existing parameters are not very convenient, and you want to add extra two parameters  to avoid affecting existing ones.

I think it is better to let another related committer check this option.  @GJL what do you think?

Reply:
maybe we need some suggestions from @GJL